src/pages/events/create.svelte

```diff
@@ -0,0 +1,162 @@
+<script>
+  import Uploader from '../../components/Uploader.svelte'
+  import { me } from '../../data/user'
+  import { events, create } from '../../data/events'
+  import { format, addDays } from 'date-fns'
+  import LocationInput from '../../components/LocationInput.svelte'
+  import Icon from 'fa-svelte'
+  import { throwError } from '../../data/error'
+  import { goto } from '@roxi/routify'
+  
+  let imageUrl = ''
+  
+  const defaults = {
+    name: '',
+    start_date: new Date(),
+    end_date: new Date(),
+  }
+  let event = {}
+  const tomorrow = format(addDays(Date.now(), 1), 'yyyy-MM-dd')
+            
+  function assertHas(value, errorMessage) {
+    if (!value) {
+      throwError(errorMessage)
+    }
+  }
+
+  function validate(event) {
+    assertHas(event.name, 'Please provide the name of the event')
+    assertHas(event.location, 'Please provide a location')
+    assertHas(event.start_date, 'Please tell us the start date')
+    assertHas(event.end_date, 'Please tell us the end date')
+  }
+  
+  async function onSubmit() {
+    validate(event)
+  
+    await create({
+      name: event.name,
+      description: event.description,
+      image_file_id: event.photo?.id,
+      location: event.location,
+      start_date: event.start_date,
+      end_date: event.end_date,
+      more_info_url: event.more_info_url,
+    })
+
+    $goto(`/events`)
+
+  }
+  
+  function imageUploaded(e) {
+    event.photo = e.detail
+    imageUrl = e.detail.url
+  }
+    
+  function onLocationChanged(e) {
+    event.location = e.detail
+  }
+
+</script>
+
+<style>
+  .preview {
+    max-height: 160px;
+  }
+</style>
+
+<h2 class="mb-3">Create an Event</h2>
+
+<form on:submit|preventDefault={onSubmit} autocomplete="off">
+  <div class="form-row form-group">
+    <label for="event-name" class="col-12 col-sm-3 col-lg-2 col-form-label-lg">
+      Event name:
+    </label>
+
+    <div class="col">
+      <input type="text" class="form-control form-control-lg" id="event-name" bind:value={event.name} placeholder="Event Name">
+    </div>
+  </div>
+
+  <div class="form-row form-group">
+    <div class="col-12 col-sm-3 col-lg-2 col-form-label-lg">
+      <label for="request-description">Description:</label>
+      <small class="text-muted font-italic">(optional)</small>
+    </div>
+    <div class="col">
+      <textarea class="form-control" bind:value={event.description} rows="3"
+                id="event-description" placeholder="Please describe the item" />
+    </div>
+  </div>
+
+  <div class="form-row form-group">
+    <div class="col-auto col-sm-3 col-lg-2 col-form-label-lg">
+      Upload image: <br />
+      <small class="text-muted font-italic">(optional)</small>
+    </div>
+
+    <div class="col-auto mt-1">
+      <Uploader on:uploaded={imageUploaded} type={ event.photo && event.photo.url ? 'change' : 'add'}/>
```

Comment:
Just as an FYI, ?. is a new operator that make this simpler. (Also on line 104)
```suggestion
      <Uploader on:uploaded={imageUploaded} type={ event.photo?.url ? 'change' : 'add'}/>
```